editor/plugins/inspector_callable_editor_plugin.cpp

```diff
@@ -0,0 +1,95 @@
+/*************************************************************************/
+/*  inspector_callable_editor_plugin.cpp                                 */
+/*************************************************************************/
+/*                       This file is part of:                           */
+/*                           GODOT ENGINE                                */
+/*                      https://godotengine.org                          */
+/*************************************************************************/
+/* Copyright (c) 2007-2022 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2022 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include "inspector_callable_editor_plugin.h"
+#include "editor/editor_node.h"
+
+bool EditorInspectorCallablePlugin::can_handle(Object *p_object) {
+	Ref<Script> scr = Object::cast_to<Script>(p_object->get_script());
+
+	if (scr == nullptr)
+		return false;
+
+	return true;
+}
+
+void EditorInspectorCallablePlugin::call_action(Object *p_object, String p_path) {
+	Ref<Script> scr = Object::cast_to<Script>(p_object->get_script());
+	if (!scr->is_tool()) {
+		print_error("Could not call action, assigned script must be set as a tool script.");
+		return;
+	}
+
+	Variant ret;
+	p_object->get_script_instance()->get(StringName(p_path), ret);
+
+	if (ret.get_type() == Variant::Type::CALLABLE) {
```

Comment:
'call_action' should not be called unless the variable is already a callable. This check isn't necessary.